src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingDiagnosticValidationPass.cs

```diff
@@ -73,5 +76,19 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
 
             return result;
         }
+
+        private class LocationIgnoringDiagnosticComparer : IEqualityComparer<RazorDiagnostic>
+        {
+            public static IEqualityComparer<RazorDiagnostic> Instance = new LocationIgnoringDiagnosticComparer();
+
+            public bool Equals(RazorDiagnostic x, RazorDiagnostic y)
+                => x is not null &&
```

Comment:
is both x and y being null not possible somehow?

Reply:
Indeed, I wouldn't expect nulls here, but more importantly I am happy to bail out if either are null, as I would think something has gone very strange, so I'm okay with dropping the formatting operation in that case.